src/spec.c

```diff
@@ -2693,6 +2748,26 @@ int IndexSpec_DeleteDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString
   return REDISMODULE_OK;
 }
 
+int IndexSpec_DeleteDoc(IndexSpec *spec, RedisModuleCtx *ctx, RedisModuleString *key) {
+  RedisSearchCtx sctx = SEARCH_CTX_STATIC(ctx, spec);
+
+  // TODO: is this necessary?
+  RedisSearchCtx_LockSpecRead(&sctx);
```

Comment:
why not lock for write directly and call the other function?

Reply:
because we might discover that we don't have to delete anything and we can save the block for all other readers (I'm not sure when this is happening but the early return was already in the code)